salt/client/ssh/__init__.py

```diff
@@ -325,6 +333,90 @@ def __init__(self, opts):
                                              python3_bin=self.opts[u'python3_bin'])
         self.mods = mod_data(self.fsclient)
 
+    def _get_roster(self):
+        '''
+        Read roster filename as a key to the data.
+        :return:
+        '''
+        roster_file = salt.roster.get_roster_file(self.opts)
+        if roster_file not in self.__parsed_rosters:
+            roster_data = compile_template(roster_file, salt.loader.render(self.opts, {}),
+                                           self.opts['renderer'], self.opts['renderer_blacklist'],
+                                           self.opts['renderer_whitelist'])
+            self.__parsed_rosters[roster_file] = roster_data
+        return roster_file
+
+    def _expand_target(self):
+        '''
+        Figures out if the target is a reachable host without wildcards, expands if any.
+        :return:
+        '''
+        # TODO: Support -L
+        target = self.opts['tgt']
+        if isinstance(target, list):
+            return
+
+        hostname = self.opts['tgt'].split('@')[-1]
+        needs_expansion = '*' not in hostname and salt.utils.network.is_reachable_host(hostname)
+        if needs_expansion:
+            hostname = salt.utils.network.ip_to_host(hostname)
+            self._get_roster()
+            for roster_filename in self.__parsed_rosters:
+                roster_data = self.__parsed_rosters[roster_filename]
+                if not isinstance(roster_data, bool):
+                    for host_id in roster_data:
+                        if hostname in [host_id, roster_data.get('host')]:
+                            if hostname != self.opts['tgt']:
+                                self.opts['tgt'] = hostname
+                            self.__parsed_rosters[self.ROSTER_UPDATE_FLAG] = False
+                            return
+
+    def _update_roster(self):
+        '''
+        Update default flat roster with the passed in information.
+        :return:
+        '''
+        roster_file = self._get_roster()
+        if os.access(roster_file, os.W_OK):
+            if self.__parsed_rosters[self.ROSTER_UPDATE_FLAG]:
+                with open(roster_file, 'a') as roster_fp:
```

Comment:
Can you change this call to use `salt.utils.files.fopen` instead of `open`?

Then lint will be happy. :)